src/logging.cc

```diff
@@ -2397,8 +2397,16 @@ const vector<string>& GetLoggingDirectories() {
     logging_directories_list = new vector<string>;
 
     if ( !FLAGS_log_dir.empty() ) {
-      // A dir was specified, we should use it
-      logging_directories_list->push_back(FLAGS_log_dir);
+      // A dir was specified, we should use it, and make sure to end with
+      // a directory delimiter.
+      const char* const dir_delim_end =
+        possible_dir_delim + sizeof(possible_dir_delim);
+      if (std::find(possible_dir_delim, dir_delim_end,
+            FLAGS_log_dir.back()) == dir_delim_end) {
```

Comment:
```suggestion
      // Ensure the specified path ends with a directory delimiter
      if (std::find(std::begin(possible_dir_delim), std::end(possible_dir_delim),
            FLAGS_log_dir.back()) == std::end(possible_dir_delim)) {
```
You may need to include `<iterator>`.

Reply:
Thanks for your review, already addressed comment.  

After testing `<iterator>` is not needed. My guess is that it is already included in an existing header file.
From https://en.cppreference.com/w/cpp/iterator/begin
It can be seen that `std::begin` and `std::end` are defined in many header files.

Reply:
That's not how it works. You do need `<iterator>` which defines `std::begin` and `std::end` for statically sized arrays  (see [IWYU](https://github.com/include-what-you-use/include-what-you-use#include-what-you-use) why relying on indirection is a bad style). By 'may' I meant to say that I did not check whether `<iterator>` is already included in the touched translation unit.

Reply:
Just checked `logging.cc`. `<iterator>` must definitely be included since it is not present there yet.

Reply:
Thanks for your further explanation. `<iterator>` has been added.